clients/src/test/java/org/apache/kafka/common/network/ChannelBuildersTest.java

```diff
@@ -79,25 +80,52 @@ public void testChannelBuilderConfigs() {
 
         // test configs with listener prefix
         Map<String, Object> configs = ChannelBuilders.channelBuilderConfigs(securityConfig, new ListenerName("listener1"));
+
         assertNull(configs.get("listener.name.listener1.gssapi.sasl.kerberos.service.name"));
+        assertFalse(securityConfig.unused().contains("listener.name.listener1.gssapi.sasl.kerberos.service.name"));
+
         assertEquals(configs.get("gssapi.sasl.kerberos.service.name"), "testkafka");
+        assertFalse(securityConfig.unused().contains("gssapi.sasl.kerberos.service.name"));
+
         assertEquals(configs.get("sasl.kerberos.service.name"), "testkafkaglobal");
+        assertFalse(securityConfig.unused().contains("gssapi.sasl.kerberos.service.name"));
```

Comment:
Should gssapi.sasl.kerberos.service.name be sasl.kerberos.service.name?

Reply:
you are right.